### PR TITLE
[18.0]MIG] stock_no_negative: Migration to 18.0

### PR DESCRIPTION
**Pull request by BertVGroenendael**

### CODE REVIEW

File: stock_no_negative/models/stock_quant.py [hunk 1]
```diff
@@ -32,7 +32,8 @@ def check_negative_qty(self):
             disallowed_by_location = not quant.location_id.allow_negative_stock
             if (
                 float_compare(quant.quantity, 0, precision_digits=p) == -1
-                and quant.product_id.type == "product"
+                and quant.product_id.is_storable
+                and quant.product_id.type == "consu"
```
Comment (rven) [inline stock_no_negative/models/stock_quant.py]: Is this line needed? is_storable is sufficient to check.
The field can only be changed when type == "consu"

**Review by rousseldenis — APPROVED** [review]

**Review by luc-adhoc — APPROVED** [review]
functional review

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Comment by moaadbourhim** [comment]
Hi @rousseldenis @pedrobaeza @rven,

I noticed that this PR does not display the "allow_negative_stock" field in the product.template.
I’ve addressed this in my PR : https://github.com/OCA/stock-logistics-workflow/pull/1766
Could you review it and let me know if there’s anything that needs adjustment?

Thank you!

**Pull request merged**